contracts/game/src/lib.cairo

```diff
@@ -224,6 +225,45 @@ mod Game {
             _start_game(ref self, weapon, name, interface_camel);
         }
 
+        fn optimistic_start(ref self: ContractState, adventurer_id: felt252, block_hash: felt252) {
+            // assert the starting entropy has not been set
+            assert(
+                self._starting_entropy.read(adventurer_id) == 0, messages::STARTING_HASH_NOT_ZERO
+            );
+
```

Comment:
lets assert hash is not zero as a basic sanity check. Contract should always do its best to protect player from a faulty client.

Reply:
this function can only be called once by the owner and only if they haven't started the game. This is what the null check does